pandas/core/missing.py

```diff
@@ -621,3 +622,24 @@ def fill_zeros(result, x, y, name, fill):
             result = result.reshape(shape)
 
     return result
+
+
+def maybe_fill(obj, value, errors):
```

Comment:
The cleanest way I can think of is writing a separate `maybe_ignore` method which factors that bit of logic out. But, you will have to run a `is_valid_fill_value` operation twice. Maybe that's OK.